File: lib/layer2_core/light/e_secondary_fixed_dim_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class ESecondaryFixedDimLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl30;
      case 'standard': return cstl30;
      case 'blue': return cbl30;
      case 'green': return cgl30;
      case 'red': return crl30;
      case 'nothing': return cnol30;
      default: return cstl30;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/ESecondaryFixedDimLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESecondaryFixedDimLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl30
            AppTheme.STANDARD -> R.color.cstl30
            AppTheme.BLUE -> R.color.cbl30
            AppTheme.GREEN -> R.color.cgl30
            AppTheme.RED -> R.color.crl30
            AppTheme.NOTHING -> R.color.cnol30
        }
    }
}
